File: walletcore/src/main/java/com/mycelium/wapi/wallet/BitcoinBasedSendRequest.java
package com.mycelium.wapi.wallet;

import com.mrd.bitlib.UnsignedTransaction;
import com.mycelium.wapi.wallet.coins.CryptoCurrency;
import com.mycelium.wapi.wallet.coins.Value;

public abstract class BitcoinBasedSendRequest<T extends GenericTransaction> extends SendRequest<T> {

   private UnsignedTransaction unsignedTx;

   protected BitcoinBasedSendRequest(CryptoCurrency type, Value fee) {
      super(type, fee);
   }

   public UnsignedTransaction getUnsignedTx() {
      return unsignedTx;
   }

}

File: walletcore/src/main/java/com/mycelium/wapi/wallet/BitcoinBasedSendRequest.kt
package com.mycelium.wapi.wallet

import com.mrd.bitlib.UnsignedTransaction
import com.mycelium.wapi.wallet.coins.CryptoCurrency
import com.mycelium.wapi.wallet.coins.Value

abstract class BitcoinBasedSendRequest<T : GenericTransaction> protected constructor(type: CryptoCurrency, fee: Value?) : SendRequest<T>(type, fee) {

    var unsignedTx: UnsignedTransaction? = null

}
